pip/req/req_install.py

```diff
@@ -312,7 +331,7 @@ def _correct_build_location(self):
         assert self._ideal_build_dir
         old_location = self._temp_build_dir
         self._temp_build_dir = None
-        new_location = self.build_location(self._ideal_build_dir)
+        new_location = self.build_location()
```

Comment:
since you now have the `_ideal_build_dir` stored as `self.req_cache.path` you can completely get rid of `self._ideal_build_dir` (and update the docstrings explaining the `_correct_build_location` dance :) )